aten/src/ATen/native/cuda/SummaryOps.cu

```diff
@@ -322,7 +322,7 @@ Tensor _histc_cuda_template(
     minvalue = minvalue - 1;
     maxvalue = maxvalue + 1;
   }
-  auto ret = cuda::CUDA_tensor_histogram<int64_t, input_t, false>(
+  auto ret = cuda::CUDA_tensor_histogram<float, input_t, false>(
```

Comment:
The output dtype should probably match the input dtype. That is what I would expect, but you should also check it against the behavior of CPU histc.